chart/templates/flower/flower-service.yaml

```diff
@@ -44,9 +44,12 @@ spec:
     component: flower
     release: {{ .Release.Name }}
   ports:
+    {{- if .Values.flower.service.portsOverride }}
+    {{- toYaml .Values.flower.service.portsOverride | nindent 4 }}
+    {{- else }}
     - name: flower-ui
-      protocol: TCP
       port: {{ .Values.ports.flowerUI }}
```

Comment:
I initially tried moving the default into values.yaml and templating it, but got stuck with port being a string instead of an int. This was the best I could come up with, but open to other options.

e.g., this doesn't work
```
# values.yaml
webserver:
  service:
    ports:
      - name: flower-ui
        port: "{{ .Values.ports.flowerUI }}"
```

Helm does have an `int`, but that only works in actual templates, not values passed through `tpl`.

Reply:
What was the error when you tried this? Did you try without the quotes?

Reply:
Without quotes:
```
Error: cannot load values.yaml: error converting YAML to JSON: yaml: invalid map key: map[interface {}]interface {}{".Values.ports.flowerUI":interface {}(nil)}
```

It needs to be a string in `values.yaml` so `tpl` works in the template itself ([like these here](https://github.com/apache/airflow/blob/79494a6f9ef491a1accdcc178e7434eb456e74da/chart/values.yaml#L1076). However, k8s will only accept an int for `port`.

One option I thought of was instead of `toYaml`ing the whole thing, range over it to `tpl` individual values.

Something like this (definitely untested):
```
{{ range .Values.webserver.service.ports }}
-
  {{ range $key, $val . }}
  {{ $key }}: {{ tpl $val }}
  {{- end }}
{{- end }}
```

Last week I dismissed it since I envisioned needing to `int` only certain keys, but that might work as-is looking at it again.